### DISCUSSION TITLE
Button to automatically update UUIDs for all users without a UUID

### DISCUSSION THREAD

**Discussion by Derkades** [NamelessMC/Nameless#2218] [Ideas]

**Comment by samerton** [comment]
The difficulty with this is Mojang's strict API rate limits - it wouldn't be as simple as looping through all users with no UUID and updating them straight away

**Reply by Derkades** [reply]
As far as I know there is no rate limit on the amount of queries, just the amount of queries for the same name/UUID

**Reply by samerton** [reply]
A few years back, a common issue with registering on Nameless sites was the username lookup failing due to rate limits - they were _very_ strict. Hopefully they've increased it since then, however I don't feel like sending potentially dozens of API requests all at once is wise

**Reply by Derkades** [reply]
I was wrong, there is a global limit of 600 requests every 10 minutes. However, you can send 10 playernames in one request so unless a website has 5000+ users with a missing UUID it is fine to update everything in one go.
https://wiki.vg/Mojang_API#Playernames_-.3E_UUIDs
